tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,516 @@
+        self.execute_command(f'routerdowngradethreshold {threshold}')
+
+    def get_max_children(self) -> int:
+        return self.__parse_int(self.execute_command('childmax'))
+
+    def set_max_children(self, val: int):
+        self.execute_command(f'childmax {val}')
+
+    #
+    # Router utilities: Child management
+    #
+    def get_child_table(self):
+        output = self.execute_command('child table')
+        if len(output) < 2:
+            raise UnexpectedCommandOutput(output)
+
+        #
+        # Example output:
+        # | ID  | RLOC16 | Timeout    | Age        | LQ In | C_VN |R|D|N| Extended MAC     |
+        # +-----+--------+------------+------------+-------+------+-+-+-+------------------+
+        # |   1 | 0x3c01 |        240 |          0 |     3 |   19 |0|0|1| be86c28929270d71 |
+        #
+        # Done
+        #
+
+        table = {}
+        for line in output[2:]:
+            line = line.strip()
+            if not line:
+                continue
+
+            fields = line.split('|')
+            _, id, rloc16, timeout, age, lq_in, c_vn, r, d, n, extaddr, _ = fields
+
+            table[int(id)] = {
+                'id': int(id),
+                'rloc16': int(rloc16, 16),
+                'timeout': int(timeout),
+                'age': int(age),
+                'lq_in': int(lq_in),
+                'c_vn': int(c_vn),
+                'r': int(r),
+                'd': int(d),
+                'n': int(n),
+                'extaddr': extaddr,
+            }
+
+        return table
+
+    def get_child_list(self):
+        line = self.__parse_str(self.execute_command(f'child list'))
+        return [int(id) for id in line.strip().split()]
+
+    #
+    # Child configurations
+    #
+
+    def get_child_timeout(self):
+        return self.__parse_int(self.execute_command('childtimeout'))
+
+    def set_child_timeout(self, timeout):
+        self.execute_command('childtimeout %d' % timeout)
+
+    def get_poll_period(self) -> float:
+        return self.__parse_int(self.execute_command('pollperiod')) / 1000.0
+
+    def set_poll_period(self, poll_period: float):
+        ms = int(poll_period * 1000)
+        self.execute_command(f'pollperiod {ms}')
+
+    #
+    # Commissioniner operations
+    #
+    def get_commissioiner_state(self):
+        return self.__parse_str(self.execute_command('commissioner state'))
+
+    def commissioner_start(self):
+        self.execute_command('commissioner start')
+
+    def commissioner_add_joiner(self, addr, psk):
+        self.execute_command('commissioner joiner add %s %s' % (addr, psk))
+
+    #
+    # Joiner operatioins
+    #
+    def joiner_start(self, psk):
+        self.execute_command('joiner start %s' % psk)
+
+    def get_joiner_id(self) -> str:
+        return self.__parse_joiner_id(self.execute_command('joiner id'))
+
+    #
+    # Allowlist management
+    #
+
+    def enable_allowlist(self):
+        self.execute_command('macfilter addr allowlist')
```

Comment:
Got error on Newman v1.52 for this command. 
# ot-ctl macfilter addr allowlist
Error 35: InvalidCommand

Reply:
I think that's because the correct command is whitelist for the openthread version on the device. 

OTCI need to check which command is available, whitelist or allowlist, and use the one that works.

Reply:
should be fixed now. Please check.